lib/remote/filterutility.cpp

```diff
@@ -114,11 +115,11 @@ bool FilterUtility::EvaluateFilter(ScriptFrame& frame, Expression *filter,
 	return Convert::ToBool(filter->Evaluate(frame));
 }
 
-static void FilteredAddTarget(ScriptFrame& permissionFrame, Expression *permissionFilter,
-	ScriptFrame& frame, Expression *ufilter, std::vector<Value>& result, const String& variableName, const Object::Ptr& target)
+static void FilteredAddTarget(ScriptFrame& permissionFrame, const Expression::Ptr& permissionFilter,
+	ScriptFrame& frame, const Expression::Ptr& ufilter, std::vector<Value>& result, const String& variableName, const Object::Ptr& target)
 {
-	if (FilterUtility::EvaluateFilter(permissionFrame, permissionFilter, target, variableName)) {
-		if (FilterUtility::EvaluateFilter(frame, ufilter, target, variableName)) {
+	if (FilterUtility::EvaluateFilter(permissionFrame, permissionFilter.get(), target, variableName)) {
+		if (FilterUtility::EvaluateFilter(frame, ufilter.get(), target, variableName)) {
```

Comment:
I personally would let this method also as-is for the same lack of need as in FilterUtility::EvaluateFilter().